code/controllers/subsystem/parallax.dm

```diff
@@ -19,32 +29,76 @@ SUBSYSTEM_DEF(parallax)
 	planet_y_offset = rand(100, 160)
 	planet_x_offset = rand(100, 160)
 
+/datum/controller/subsystem/parallax/Initialize(start_timeofday)
+	. = ..()
+	RegisterSignal(SSdcs, COMSIG_GLOB_MOB_LOGGED_IN, .proc/on_mob_login)
+	throttle_ghost_pop = CONFIG_GET(number/parallax_ghost_disable_pop)
+	throttle_all_pop = CONFIG_GET(number/parallax_disable_pop)
+
 /datum/controller/subsystem/parallax/fire(resumed = 0)
-	if (!resumed)
-		src.currentrun = GLOB.clients.Copy()
-
-	//cache for sanic speed (lists are references anyways)
-	var/list/currentrun = src.currentrun
-
-	while(length(currentrun))
-		var/client/C = currentrun[currentrun.len]
-		currentrun.len--
-		if (!C || !C.eye)
-			if (MC_TICK_CHECK)
-				return
-			continue
-		var/atom/movable/A = C.eye
-		if(!istype(A))
+	//Swap the 2 lists
+	if(!length(currentrun))
+		//Nothing to process here
+		if(!length(queued))
+			return
+		var/temp = currentrun
+		currentrun = queued
+		queued = temp
+		current_run_pointer = 1
+		//Check client count
+		throttle_ghosts = throttle_ghost_pop && length(GLOB.clients) >= throttle_ghost_pop
+		throttle_all = throttle_all_pop && length(GLOB.clients) >= throttle_all_pop
```

Comment:
The check for length clients can be merged

Reply:
I don't understand what you mean, these 2 things use different variables entirely and can't be merged

Reply:
I mean that you are checking len of GLOB.clients twice

Reply:
nitpicky but if we really want to optimize it, yes, `var/client_len = length(GLOB.clients)` would be ideal